File: sdkwork-app-sdk-typescript/src/api/drive.ts
import { appApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { DriveBatchDeleteForm, DriveContentUpdateForm, DriveCopyForm, DriveFolderCreateForm, DriveMoveForm, DriveRenameForm, PlusApiResultDriveContentVO, PlusApiResultDriveItemDetailVO, PlusApiResultDriveItemVO, PlusApiResultPageDriveItemVO, PlusApiResultVoid } from '../types';


export class DriveApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Restore drive item */
  async restoreItem(itemId: string | number): Promise<PlusApiResultDriveItemVO> {
    return this.client.put<PlusApiResultDriveItemVO>(appApiPath(`/drive/items/${itemId}/restore`));
  }

/** Rename drive item */
  async renameItem(itemId: string | number, body: DriveRenameForm): Promise<PlusApiResultDriveItemVO> {
    return this.client.put<PlusApiResultDriveItemVO>(appApiPath(`/drive/items/${itemId}/rename`), body);
  }

/** Move drive item */
  async moveItem(itemId: string | number, body: DriveMoveForm): Promise<PlusApiResultDriveItemVO> {
    return this.client.put<PlusApiResultDriveItemVO>(appApiPath(`/drive/items/${itemId}/move`), body);
  }

/** Get drive file content */
  async getItemContent(itemId: string | number): Promise<PlusApiResultDriveContentVO> {
    return this.client.get<PlusApiResultDriveContentVO>(appApiPath(`/drive/items/${itemId}/content`));
  }

/** Update drive file content */
  async updateItemContent(itemId: string | number, body: DriveContentUpdateForm): Promise<PlusApiResultDriveContentVO> {
    return this.client.put<PlusApiResultDriveContentVO>(appApiPath(`/drive/items/${itemId}/content`), body);
  }

/** Archive drive item */
  async archiveItem(itemId: string | number): Promise<PlusApiResultDriveItemVO> {
    return this.client.put<PlusApiResultDriveItemVO>(appApiPath(`/drive/items/${itemId}/archive`));
  }

/** Favorite drive item */
  async favoriteItem(itemId: string | number): Promise<PlusApiResultDriveItemVO> {
    return this.client.post<PlusApiResultDriveItemVO>(appApiPath(`/drive/items/${itemId}/favorite`));
  }

/** Unfavorite drive item */
  async unfavoriteItem(itemId: string | number): Promise<PlusApiResultDriveItemVO> {
    return this.client.delete<PlusApiResultDriveItemVO>(appApiPath(`/drive/items/${itemId}/favorite`));
  }

/** Copy drive item */
  async copyItem(itemId: string | number, body: DriveCopyForm): Promise<PlusApiResultDriveItemVO> {
    return this.client.post<PlusApiResultDriveItemVO>(appApiPath(`/drive/items/${itemId}/copy`), body);
  }

/** Batch delete drive items */
  async batchDeleteItems(body: DriveBatchDeleteForm): Promise<PlusApiResultVoid> {
    return this.client.post<PlusApiResultVoid>(appApiPath(`/drive/items/batch-delete`), body);
  }

/** Create drive folder */
  async createFolder(body: DriveFolderCreateForm): Promise<PlusApiResultDriveItemVO> {
    return this.client.post<PlusApiResultDriveItemVO>(appApiPath(`/drive/folders`), body);
  }

/** List drive items */
  async listItems(params?: QueryParams): Promise<PlusApiResultPageDriveItemVO> {
    return this.client.get<PlusApiResultPageDriveItemVO>(appApiPath(`/drive/items`), params);
  }

/** Get drive item detail */
  async getItemDetail(itemId: string | number): Promise<PlusApiResultDriveItemDetailVO> {
    return this.client.get<PlusApiResultDriveItemDetailVO>(appApiPath(`/drive/items/${itemId}`));
  }

/** Delete drive item */
  async deleteItem(itemId: string | number): Promise<PlusApiResultVoid> {
    return this.client.delete<PlusApiResultVoid>(appApiPath(`/drive/items/${itemId}`));
  }
}

export function createDriveApi(client: HttpClient): DriveApi {
  return new DriveApi(client);
}

File: sdkwork-app-sdk-go/api/drive.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/app-sdk/types"
    sdkhttp "github.com/sdkwork/app-sdk/http"
)

type DriveApi struct {
    client *sdkhttp.Client
}

func NewDriveApi(client *sdkhttp.Client) *DriveApi {
    return &DriveApi{client: client}
}

// Restore drive item
func (a *DriveApi) RestoreItem(itemId string) (sdktypes.PlusApiResultDriveItemVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/drive/items/%s/restore", itemId)), nil, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemVO](raw)
}

// Rename drive item
func (a *DriveApi) RenameItem(itemId string, body sdktypes.DriveRenameForm) (sdktypes.PlusApiResultDriveItemVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/drive/items/%s/rename", itemId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemVO](raw)
}

// Move drive item
func (a *DriveApi) MoveItem(itemId string, body sdktypes.DriveMoveForm) (sdktypes.PlusApiResultDriveItemVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/drive/items/%s/move", itemId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemVO](raw)
}

// Get drive file content
func (a *DriveApi) GetItemContent(itemId string) (sdktypes.PlusApiResultDriveContentVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/drive/items/%s/content", itemId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDriveContentVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveContentVO](raw)
}

// Update drive file content
func (a *DriveApi) UpdateItemContent(itemId string, body sdktypes.DriveContentUpdateForm) (sdktypes.PlusApiResultDriveContentVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/drive/items/%s/content", itemId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDriveContentVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveContentVO](raw)
}

// Archive drive item
func (a *DriveApi) ArchiveItem(itemId string) (sdktypes.PlusApiResultDriveItemVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/drive/items/%s/archive", itemId)), nil, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemVO](raw)
}

// Favorite drive item
func (a *DriveApi) FavoriteItem(itemId string) (sdktypes.PlusApiResultDriveItemVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/drive/items/%s/favorite", itemId)), nil, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemVO](raw)
}

// Unfavorite drive item
func (a *DriveApi) UnfavoriteItem(itemId string) (sdktypes.PlusApiResultDriveItemVO, error) {
    raw, err := a.client.Delete(AppApiPath(fmt.Sprintf("/drive/items/%s/favorite", itemId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemVO](raw)
}

// Copy drive item
func (a *DriveApi) CopyItem(itemId string, body sdktypes.DriveCopyForm) (sdktypes.PlusApiResultDriveItemVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/drive/items/%s/copy", itemId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemVO](raw)
}

// Batch delete drive items
func (a *DriveApi) BatchDeleteItems(body sdktypes.DriveBatchDeleteForm) (sdktypes.PlusApiResultVoid, error) {
    raw, err := a.client.Post(AppApiPath("/drive/items/batch-delete"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultVoid
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVoid](raw)
}

// Create drive folder
func (a *DriveApi) CreateFolder(body sdktypes.DriveFolderCreateForm) (sdktypes.PlusApiResultDriveItemVO, error) {
    raw, err := a.client.Post(AppApiPath("/drive/folders"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemVO](raw)
}

// List drive items
func (a *DriveApi) ListItems(query map[string]interface{}) (sdktypes.PlusApiResultPageDriveItemVO, error) {
    raw, err := a.client.Get(AppApiPath("/drive/items"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPageDriveItemVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPageDriveItemVO](raw)
}

// Get drive item detail
func (a *DriveApi) GetItemDetail(itemId string) (sdktypes.PlusApiResultDriveItemDetailVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/drive/items/%s", itemId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDriveItemDetailVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDriveItemDetailVO](raw)
}

// Delete drive item
func (a *DriveApi) DeleteItem(itemId string) (sdktypes.PlusApiResultVoid, error) {
    raw, err := a.client.Delete(AppApiPath(fmt.Sprintf("/drive/items/%s", itemId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVoid
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVoid](raw)
}
